drivers/storage/portworx/util/util.go

```diff
@@ -216,6 +216,18 @@ func IsPortworxEnabled(cluster *corev1.StorageCluster) bool {
 	return err != nil || !disabled
 }
 
+// IsCSIEnabled returns true if CSI is not disabled by the feature flag
+func IsCSIEnabled(cluster *corev1.StorageCluster) bool {
+	enableCSI := true
+	if FeatureCSI.IsSet(cluster.Spec.FeatureGates) {
+		enableCSI = FeatureCSI.IsEnabled(cluster.Spec.FeatureGates)
+	}
+
+	// Only enable CSI when Portworx is enabled
+	enableCSI = IsPortworxEnabled(cluster) && enableCSI
+	return enableCSI
```

Comment:
nit: The `feature.IsEnabled` method in feature.go could be simply written like this - 
```
func (feature Feature) IsEnabled(featureMap map[string]string) bool {
	enabled, err := strconv.ParseBool(featureMap[string(feature)])
	return err != nil || enabled
}
```
The above also makes sure, if the user has not properly disabled (by giving non-boolean or empty value) the feature, then we still return true.

Then we can simply write this helped method as follows - 
```
return IsPortworxEnabled(cluster) && FeatureCSI.IsEnabled(cluster.Spec.FeatureGates)
```